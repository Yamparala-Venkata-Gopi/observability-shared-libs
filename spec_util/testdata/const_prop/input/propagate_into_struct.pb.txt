# MethodTemplate proto

method_id: {
  name: "new_method_with_refs"
}

arg_templates: {
  key: "arg1"
  value: {
    struct_template: {
      field_templates: {
        # This field should become a constant because it refers to a constant.
        key: "field_with_ref_to_const"
        value: {
          ref: {
            method_index: 0
            arg_ref: {
              key: "primitive_constant_arg"
              data_ref: {
                primitive_ref: {}
              }
            }
          }
        }
      }
      field_templates: {
        key: "field_with_ref_to_response"
        value: {
          ref: {
            method_index: 0
            response_ref: {
              key: "response_1"
              data_ref: {
                primitive_ref: {}
              }
            }
          }
        }
      }
    }
  }
}
