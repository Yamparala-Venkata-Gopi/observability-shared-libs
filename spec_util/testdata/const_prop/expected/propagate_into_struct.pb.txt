# MethodTemplate proto

method_id: {
  name: "new_method_with_refs"
}

arg_templates: {
  key: "arg1"
  value: {
    struct_template: {
      field_templates: {
        key: "field_with_ref_to_const"
        value: {
          value: {
            primitive: {
              type_hint: "string_constant_1"
              string_value: {
                value: "7c73565e-0d62-49b0-84e7-3063f4d53d9c"
              }
            }
          }
        }
      }
      field_templates: {
        key: "field_with_ref_to_response"
        value: {
          ref: {
            method_index: 0
            response_ref: {
              key: "response_1"
              data_ref: {
                primitive_ref: {}
              }
            }
          }
        }
      }
    }
  }
}
